pkg/controller/controller.go

```diff
@@ -1370,6 +1443,144 @@ func (c *controller) bindingDelete(obj interface{}) {
+		glog.Warning(s)
+		c.updateInstanceCondition(
+			instance,
+			v1alpha1.InstanceConditionReady,
+			v1alpha1.ConditionFalse,
+			"ReferencesNonexistentServicePlan",
+			"The instance references a ServicePlan that does not exist. "+s,
+		)
+		c.recorder.Event(instance, api.EventTypeWarning, errorNonexistentServicePlanReason, s)
+		return nil, nil, "", nil, fmt.Errorf(s)
+	}
+
+	broker, err := c.brokerLister.Get(serviceClass.BrokerName)
+	if err != nil {
+		s := fmt.Sprintf("Instance \"%s/%s\" references a non-existent broker %q", instance.Namespace, instance.Name, serviceClass.BrokerName)
+		glog.Warning(s)
+		c.updateInstanceCondition(
+			instance,
+			v1alpha1.InstanceConditionReady,
+			v1alpha1.ConditionFalse,
+			errorNonexistentBrokerReason,
+			"The instance references a Broker that does not exist. "+s,
+		)
+		c.recorder.Event(instance, api.EventTypeWarning, errorNonexistentBrokerReason, s)
+		return nil, nil, "", nil, err
+	}
+
+	username, password, err := getAuthCredentialsFromBroker(c.kubeClient, broker)
+	if err != nil {
+		s := fmt.Sprintf("Error getting broker auth credentials for broker %q: %s", broker.Name, err)
+		glog.Info(s)
+		c.updateInstanceCondition(
+			instance,
+			v1alpha1.InstanceConditionReady,
+			v1alpha1.ConditionFalse,
+			errorAuthCredentialsReason,
+			"Error getting auth credentials. "+s,
+		)
+		c.recorder.Event(instance, api.EventTypeWarning, errorAuthCredentialsReason, s)
+		return nil, nil, "", nil, err
+	}
+
+	glog.V(4).Infof("Creating client for Broker %v, URL: %v", broker.Name, broker.Spec.URL)
+	brokerClient := c.brokerClientCreateFunc(broker.Name, broker.Spec.URL, username, password)
+	return serviceClass, servicePlan, broker.Name, brokerClient, nil
+}
+
+// getServiceClassPlanAndBrokerForBinding is a sequence of operations that's
+// done to validate service plan, service class exist, and handles creating
+// a brokerclient to use for a given Instance.
+func (c *controller) getServiceClassPlanAndBrokerForBinding(instance *v1alpha1.Instance, binding *v1alpha1.Binding) (*v1alpha1.ServiceClass, *v1alpha1.ServicePlan, string, brokerapi.BrokerClient, error) {
+	serviceClass, err := c.serviceClassLister.Get(instance.Spec.ServiceClassName)
+	if err != nil {
+		s := fmt.Sprintf("Binding \"%s/%s\" references a non-existent ServiceClass %q", binding.Namespace, binding.Name, instance.Spec.ServiceClassName)
+		glog.Warning(s)
+		c.updateBindingCondition(
+			binding,
+			v1alpha1.BindingConditionReady,
+			v1alpha1.ConditionFalse,
+			errorNonexistentServiceClassReason,
+			"The binding references a ServiceClass that does not exist. "+s,
+		)
+		c.recorder.Event(binding, api.EventTypeWarning, "ReferencesNonexistentServiceClass", s)
+		return nil, nil, "", nil, err
+	}
+
+	servicePlan := findServicePlan(instance.Spec.PlanName, serviceClass.Plans)
```

Comment:
No, because it updates the bindings condition instead of the instance. There's probably a way to combine the two in a clever manner, but for now they are separate.